docs/pages/reference/js/dscvr/index.mdx

```diff
@@ -0,0 +1,224 @@
+import { HomePage } from 'vocs/components';
+
+# DSCVR Support
+
+Cross-protocol frames are supported by frames.js via familiar APIs. This guide will showcase how to write a simple stateless frame, which returns the identity of the user that interacted with the frame in Farcaster or DSCVR.
+
+## Setup
+
+First, you need to install `frames.js` and `@dscvr-one/frames-adapter`. You can do this by running the following command:
+
+<HomePage.InstallPackage
+  name="frames.js @dscvr-one/frames-adapter"
+  type="install"
+/>
+
+## Writing a Frame
+
+To write a frame using Next.js, we need to create a page which renders the frame and a route which handles frame action requests.
+
+### API Route
+
+We start by creating a new API route which will handle POST requests to our frame. In your Next.js project, create a new directory `/frames` and inside it, create a `route.ts` file which contains the following code:
+
+```tsx filename="example0.ts"
+export { POST } from 'frames.js/next/server';
+```
+
+This will handle POST requests to our frame and redirect them to page we are about to create, which will handle the rendering logic.
+
+### Page
+
+In your Next.js project, create a new `page.tsx` at the root of the project and write the following code:
+
+First import the necessary functions and components from `frames.js`:
+
+```tsx filename="example1.tsx"
+import {
+  FrameButton,
+  FrameContainer,
+  FrameImage,
+  NextServerPageProps,
+  getFrameMessage,
+  getPreviousFrame,
+} from 'frames.js/next/server';
+```
+
+`FrameButton`, `FrameContainer`, and `FrameImage` are components that are used to construct Frame metadata tags in HTML. `NextServerPageProps` is a type that you can use to define the props of your page. `getFrameMessage` and `getPreviousFrame` are functions that you can use to get the message and the previous frame of your page to determine the next state to return.
+
+Next, import the DSCVR validation methods from frames.js
+
+```tsx filename="example2.tsx"
+// [!include example1.tsx]
+import {
+  getDscvrFrameMessage,
+  isDscvrFrameActionPayload,
+} from 'frames.js/dscvr'; // [!code focus]
```

Comment:
```suggestion
} from '@dscvr-one/frames.js/dscvr'; // [!code focus]
```

Reply:
Right now it would not work like this, if you try to install `npm i @dscvr-one/frames.js/dscvr` you get an error.
Eventually when we merged with frames.js, out package will be at `frames.js/dscvr` the same as `frames.js/xmtp`

Reply:
Thanks, done